Author: Jason Carey
Description: Fix call to vec_extract for ppc64el
Origin: https://jira.mongodb.org/browse/SERVER-33395
Last-Update: 2018-03-23
--- a/src/mongo/db/fts/unicode/byte_vector_altivec.h
+++ b/src/mongo/db/fts/unicode/byte_vector_altivec.h
@@ -98,7 +98,7 @@
         // big endian by comparison.
         const Native bits = {120, 112, 104, 96, 88, 80, 72, 64, 56, 48, 40, 32, 24, 16, 8, 0};
 
-        return vec_extract(vec_vbpermq(_data, bits), 0);
+        return vec_extract(vec_vbpermq(_data, bits), 1);
     }
 
     /**
